### ISSUE TITLE
Cable source

### ISSUE THREAD

**Issue by kydar1** [Novgorod/LANC-USB-GUI#2]
Where did you source that LANC cable, if you don't mind me asking?

**Comment by Novgorod** [comment]
I made the cable myself because it's an adapter between a LANC port and the Arduino. If you can manage to solder up the transistor/diode circuit for it, you'll have no problems making the cable. Use any 3-wire signal cable and a 2.5mm mini "stereo" (or TRS) jack on the LANC side - NOT the more common 3.5mm headphone jack. The Arduino side has standard Dupont connectors (2.54mm pitch or single pins) to slide on the pins; or you can solder the cable directly to the adapter board with the transistor. All parts are available from any electronic components shop.

The 2.5mm stereo jack is de-facto standard for LANC ports on video cameras and accessories, but some older devices (especially tape decks) may have other LANC connector formats, such as DIN plugs or proprietary plugs, so make sure to get the right plug for your device.

**Comment by kydar1** [comment]
Thanks for that.  Hey I'd really like to build one of these but I have
pretty limited knowledge and experience with small electronics so I'm
wondering if you can help me out.  I found some generic 2.5mm cables/jacks
and ordered one, and the Arduino is pretty easy to source too, but I kind
of get lost trying to follow your diagrams on Github.  It seems I need 2
resistors but I don't understand the nomenclature.  Does 5K6 mean 5600
ohms, and 4K7 = 4700?  And I need a BC547 transistor, the only ones I can
find are BC547 TO-92 and I'm not sure they are exactly the same.  Not sure
where to get a Zener diode either.  There are no local electronics stores
anymore.  Finally, what do you call that piece that the transistor is
soldered to?  Could you possibly take a few close-up photos of it showing
how the components are arranged on it and how it connects to the Arduino?
I greatly appreciate your help.  I can tip if you provide a Venmo/Paypal
account.  Tell me how much you think is reasonable.

**Comment by Novgorod** [comment]
You got it right, the resistors are 4700 and 5600 Ohms (the number after the "k" is the first decimal place, so 4k7 means 4k + 0.7k). As for the transistor, TO-92 refers to the package or shape, that's exactly the one I use (small D-shape with through-hole legs; other packages are for example SMD). As long as it's a "BC547" model they will perform identically. The Zener diode should have a 5.1V value ("5V1").

I'm not sure what you mean by where's the transistor soldered to - the whole circuit (transistor, diode, resistors, connectors) is built on a small piece of prototyping board, but you can also use a breadboard to build and test the circuit, then it's more clunky but you don't have to solder anything. Which (online) electronics store do you want to get the parts from? I could look up the components...

**Comment by kydar1** [comment]
Yes that's it, the prototyping board, that has the male and female pins to
connect to both the Arduino and the LANC cable (definitely buying the
Arduino version that comes with headers/pins).  Can't seem to find one
similar.  I was trying to look at the photo you posted up close but the
resolution isn't really high enough, to see how you soldered and connected
the resistors, transistor, and diode.  Are you in USA?  Amazon sells lots
of different "electronics fun kits" that have breadboards, jumper wires,
and various resistors, diodes, etc.  Could probably find one that had
everything except the BC547 if I looked hard enough.

Many thanks to you for developing this code which, once I get all the
hardware and software components together, I know is going to work great
for capturing all that metadata from my old Hi8 tapes.  The Walter Berndl
method seems primitive in comparison to what you accomplished.

**Comment by Novgorod** [comment]
Thanks for the praises, but you still have to get it to work :) ...

I'm in the EU, but in the US you have huge electronics shops like Digikey where you can get everything. Maybe the Amazon kits will be fine too, though a 5V Zener diode seems a bit exotic for that. Here are the original photos of the adapter board and some more angles: https://photos.app.goo.gl/kFDDU4YmsMPs1ryB9

Note the 2 pins stuck to the bottom of the board - you DON'T need them, they are just a convenience feature for me to connect an oscilloscope to the frame sync signal from the LANC device. Good luck, hope you get it to work...

**Comment by kydar1** [comment]
I have looked over your photos and schematics thoroughly but I'm missing
something, I can't figure it out.  There seem to be several solder points
underneath that are connected to nothing.  For example, the emitter from
the transistor connects to a long solder bead underneath that goes to
nothing.  It doesn't match the schematic.  Also I think (not 100% sure)
that you mislabeled the transistor pins on the schematic as EBC when in
fact they should be CBE.

**Comment by Novgorod** [comment]
The emitter is right at the edge of the little board and is connected to ground. There is a solder trace going all the way to the other side of the board. What's not so clear is that the entire hole column on the right edge (next to the LANC cable connector) is connected together on this prototype board (the copper trace forms a continuous path across that hole column):
![image](https://github.com/user-attachments/assets/bc4756e3-2d97-45f7-8bc3-3ac90500355d)

And the E/B/C labeling on the schematic shows a BOTTOM view of the transistor where you're looking at the legs.

**Comment by kydar1** [comment]
Are you saying that the printed light lines on the board represent
continuous traces?  I did not realize that.  If that is the case though,
then when I look at the 4.7k resistor, it appears to be connected to
adjacent holes that are continuous, therefore shorted.  Please realize that
while I'm a reasonably intelligent fellow (or at least I like to think so)
I just don't have any experience with electronics design.

By the way, do you ever sleep? 😂

**Comment by Novgorod** [comment]
No, the white lines on the top side are just some raster. On this little PCB section ONLY the right column of holes next to the LANC pins is continuous, all other holes are isolated. Some prototype boards have weird patterns such as continuous lines around the edges (like the one where this section was cut from). I'd suggest to get a simple hole pattern without any connected lines or it will get confusing...

I don't sleep much, but I'm on metric time, so it's just dinner time around here, no worries :) ..

**Comment by kydar1** [comment]
From the photo you just sent me, it appears that the collector pin is
soldered through the middle row, directly to the left of the yellow LANC
wire, whereas the base and emitter are in the next row, to the left of the
red wire.  Is that correct?  It was difficult to tell from the other photos
you sent earlier.

**Comment by Novgorod** [comment]
Yes, the collector leg is bent a bit so it goes into the middle row while the other two are one row away from the bottom edge.

**Comment by kydar1** [comment]
Ok, I am finally beginning to understand.  I found the following on Amazon,
can you kindly give your opinion whether these are suitable parts?

Diode https://www.amazon.com/dp/B081MQSYR5
Transistor https://www.amazon.com/dp/B0CYZD57ZG
Resistors https://www.amazon.com/dp/B07P3MFG5D (package contains both 4.7k
and 5.6k)
Prototype board https://www.amazon.com/dp/B07ZYNWJ1S


Parts that I am not sure about are:

1. The long rectangular connectors that it seems you soldered the LANC
wires into
2. The small square connectors with the 90° tops that are soldered onto the
prototype board, that the LANC wires connect to.
3. The 1x5 black connector that plugs into the Arduino.

I could probably find these parts on Amazon or Digikey, I just don't know
what they are properly called so I don't know how to search for them.

**Comment by Novgorod** [comment]
The Amazon parts look fine. As for the prototype board, take note whether it has connected rows/columns (some of them look like they have continuous power/ground rails at the edges) - it's not an issue as long as you're aware.

The 3 other items are so-called Dupont connectors (male/female, cable/solder, 90°/straight variants) with 2.54mm pitch. There are kits like this: https://www.amazon.com/dp/B08X6C7PZM/
A whole set like that is overkill for one project, but if you look for "Dupont connector" with some other keywords (like wire, solder, angle etc.), you can find small quantities in the right format.
